http-body-util/src/stream.rs

```diff
@@ -0,0 +1,76 @@
+use bytes::Buf;
+use futures_util::stream::{BoxStream, Stream};
+use http::HeaderMap;
+use http_body::Body;
+use std::{
+    fmt,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+/// A body created from a `Stream`.
+pub struct StreamBody<D, E> {
+    stream: BoxStream<'static, Result<D, E>>,
```

Comment:
Actually I was thinking it might make sense to keep the stream as a generic type rather than always boxing it.

Then users can box the body if they need it but aren't forced to for example if returning `impl IntoResponse` when using axum.